moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common.h

```diff
@@ -0,0 +1,555 @@
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2008, Willow Garage, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: E. Gil Jones */
+
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/utils/robot_model_test_utils.h>
+#include <moveit/collision_detection/collision_common.h>
+#include <moveit/collision_detection/collision_world.h>
+#include <moveit/collision_detection/collision_robot.h>
+#include <moveit/collision_detection/collision_detector_allocator.h>
+#include <urdf_parser/urdf_parser.h>
+#include <geometric_shapes/shape_operations.h>
+
+#include <gtest/gtest.h>
+#include <sstream>
+#include <algorithm>
+#include <ctype.h>
+#include <fstream>
+
+template <class T>
+class CollisionDetectorTest : public ::testing::Test
+{
+public:
+  T value_;
+
+protected:
+  CollisionDetectorTest()
+  {
+  }
+
+  void SetUp() override
+  {
+    robot_model_ = moveit::core::loadTestingRobotModel("pr2");
+    robot_model_ok_ = static_cast<bool>(robot_model_);
+    kinect_dae_resource_ = "package://moveit_resources/pr2_description/urdf/meshes/sensors/kinect_v0/kinect.dae";
+
+    acm_.reset(new collision_detection::AllowedCollisionMatrix(robot_model_->getLinkModelNames(), true));
+
+    collision_detection::CollisionDetectorAllocatorPtr alloc(new T);
```

Comment:
Do you need a `new T` here, or can you reference the pointer to `value_`, i.e. `alloc(&value_);`?

Reply:
You're right